src/app/children/child-details/child-details.component.ts

```diff
@@ -89,26 +103,67 @@ export class ChildDetailsComponent implements OnInit {
 
   // TODO Anlegen einer eigenen Klasse socialworker als subklasse von User
   socialworkers: String[];
+  form: FormGroup;
 
-  constructor(private entityMapperService: EntityMapperService, private route: ActivatedRoute) {
+
+  initializeForm(){
+    this.form = this.fb.group({
+      name: [{value: this.child.name, disabled:!this.editable}, Validators.required],
+      gender: [{value: this.child.gender, disabled:!this.editable}],
+      project: [{value: this.child.pn, disabled:!this.editable}],
+      birthday: [{value: this.child.dateOfBirth, disabled:!this.editable}],
+      motherTongue: [{value: this.child.motherTongue, disabled:!this.editable}],
+      admission: [{value: this.child.admission, disabled:!this.editable}],
+      religion: [{value: this.child.religion, disabled:!this.editable}]
+    });
+  }
 
 
+  constructor(private entityMapperService: EntityMapperService, private route: ActivatedRoute,
+              @Inject(FormBuilder) public fb: FormBuilder, public router: Router) {
+
+    let id = this.route.snapshot.params['id'];
+    if (id == "new") {
+      this.creating = true;
+      this.editable = true;
+      let uniqid = require('uniqid');
+      console.log("id: " + uniqid);
+      this.child = new Child(uniqid());
+    } else {
+      this.entityMapperService.load<Child>(Child, id)
+        .then(child => {
+          this.child = child;
+          this.initializeForm();
+        })
+        .catch(err => console.log(err + "load2"));
+    }
+    this.initializeForm();
   }
 
   switchEdit() {
     this.editable = !this.editable;
+    this.initializeForm();
   }
 
-  ngOnInit() {
-    const params = this.route.snapshot.params;
-    const childId = params['id'];
-
-    this.entityMapperService.load<Child>(Child, childId)
-      .then(child => this.child = child);
-    // this.editable = false;
-    this.genderSelector = Object.keys(this.gender).filter(k => !isNaN(Number(k)));
-    this.selectedGender = '0';
+  save() {
+    this.child.name = this.form.get("name").value;
```

Comment:
wouldn't it be possible to avoid all these assignments using Angular's two-way data binding? https://angular.io/guide/template-syntax#binding-syntax-an-overview

Reply:
when you use 2 way bindings, you always override the loaded child, so there is not easy way to cancel the changes you made. This way you have a layer in between and we also can define which of the fields are optional and which are necessary for creation/editing

Reply:
ah, that's interesting. This together with `initializeForm()` looks like a lot of boilerplate code but maybe it's a good solution.
I tried to solve the same issue at other places: https://github.com/NGO-DB/ndb-core/blob/864fbf6a45d09dfec889651ae4ed2e7e78c92302/src/app/ui-helper/entity-subrecord/entity-subrecord.component.ts#L59-L69 and https://github.com/NGO-DB/ndb-core/blob/864fbf6a45d09dfec889651ae4ed2e7e78c92302/src/app/children/notes/note-details/note-details.component.ts#L47  Not sure what's the best way overall ...